datafusion/core/src/physical_optimizer/sanity_checker.rs

```diff
@@ -0,0 +1,152 @@
+        })
+        .data()
+    }
+
+    fn name(&self) -> &str {
+        "SanityCheckPlan"
+    }
+
+    fn schema_check(&self) -> bool {
+        true
+    }
+}
+
+pub fn check_sort_requirements(
+    plan: Arc<dyn ExecutionPlan>,
+) -> Result<Transformed<Arc<dyn ExecutionPlan>>> {
+    let children = plan.children();
+    let sort_reqs = plan.required_input_ordering();
+    let children_len = children.len();
+    // TODO: Use izip!.
+    for i in 0..children_len {
+        let child = &children[i];
+        let child_sort_req = &sort_reqs[i];
+
+        // No requirement for the child
+        if child_sort_req.is_none() {
+            continue;
+        }
+
+        let child_sort_req = child_sort_req.as_ref().unwrap();
+        if !child
+            .equivalence_properties()
+            .ordering_satisfy_requirement(&child_sort_req)
+        {
+            return plan_err!(
+                "Child: {:?} does not satisfy parent order requirements",
+                child
+            );
+        }
+    }
+
+    Ok(Transformed::no(plan))
+}
+
+pub fn check_partition_requirements(
+    plan: Arc<dyn ExecutionPlan>,
+) -> Result<Transformed<Arc<dyn ExecutionPlan>>> {
+    Ok(Transformed::no(plan))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use crate::physical_optimizer::test_utils::{
+        bounded_window_exec, memory_exec, sort_exec, sort_expr_options
+    };
+    use arrow::compute::SortOptions;
+    use arrow::datatypes::{DataType, Field, Schema, SchemaRef};
+    use datafusion_common::Result;
+
+    fn create_test_schema() -> Result<SchemaRef> {
+        let c9_column = Field::new("c9", DataType::Int32, true);
+        let schema = Arc::new(Schema::new(vec![c9_column]));
+        Ok(schema)
+    }
+
+    #[tokio::test]
+    async fn test_bw_sort_requirement() -> Result<()> {
+        let schema = create_test_schema()?;
+        let source = memory_exec(&schema);
+        let sort_exprs = vec![sort_expr_options(
+            "c9",
+            &source.schema(),
+            SortOptions {
+                descending: false,
+                nulls_first: false,
+            },
+        )];
+        let sort = sort_exec(sort_exprs.clone(), source);
+        let bw = bounded_window_exec("c9", sort_exprs, sort);
+        let sanity_checker = SanityCheckPlan::new();
+        let opts = ConfigOptions::default();
+        assert_eq!(sanity_checker.optimize(bw, &opts).is_ok(), true);
+        Ok(())
+    }
+
+    #[tokio::test]
+    async fn test_bw_no_sort_requirement() -> Result<()> {
+        let schema = create_test_schema()?;
+        let source = memory_exec(&schema);
+        let sort_exprs = vec![sort_expr_options(
+            "c9",
+            &source.schema(),
+            SortOptions {
+                descending: false,
+                nulls_first: false,
+            },
+        )];
+        let bw = bounded_window_exec("c9", sort_exprs, source);
```

Comment:
Same thing applies here

Reply:
I have added plan checks in the below commit:

https://github.com/synnada-ai/datafusion-upstream/pull/24/commits/aea95c66f7fe59826ea435597b2009a24bebf897

Reply:
👍